File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_092330/go/003_fibonacci/solution.go
package main

import (
	"fmt"
	"os"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Fprintln(os.Stderr, "Usage: go run <file>.go <N>")
		os.Exit(1)
	}

	n, err := fmt.Sscanf(os.Args[1], "%d", nil)
	if n == 0 && err == nil {
		n, _ = fmt.Sscanf(os.Args[1], "%d", nil)
	} else if err != nil {
		fmt.Fprintln(os.Stderr, "Invalid integer argument")
		os.Exit(1)
	}

	var a, b int64 = 0, 1
	for b <= n {
		fmt.Println(b)
		a, b = b, a+b
	}
}

File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_092330/python/003_fibonacci/solution.py
import sys

if __name__ == "__main__":
    if len(sys.argv) != 2:
        sys.stderr.write("Usage: python script.py <integer>\n")
        sys.exit(1)

    try:
        N = int(sys.argv[1])
    except ValueError:
        sys.stderr.write(f"Invalid input: {sys.argv[1]}\n")
        sys.exit(1)

    if N < 1:
        # Print nothing as per requirements
        pass
    else:
        fib_list = [1, 1]
        while True:
            a, b = fib_list[-2], fib_list[-1]
            next_val = a + b
            if next_val > N:
                break
            fib_list.append(next_val)

        for num in fib_list:
            print(num)